src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -435,6 +460,36 @@ def infer_expr(
       having a domain argument now, and a dictionary mapping symbol names referenced in `expr` to
       domain they are accessed at.
     """
+
+    itir_type_inference.reinfer(
+        expr, offset_provider_type=common.offset_provider_to_type(offset_provider)
+    )
+    domain, el_types = _canonicalize_domain_structure(
+        domain,
+        gtx_utils.tree_map(collection_type=ts.TupleType, result_collection_constructor=tuple)(
+            lambda x: x
+        )(expr.type),
+    )
+
+    if cpm.is_applied_as_fieldop(expr) and cpm.is_call_to(expr.fun.args[0], "scan"):  # type: ignore[attr-defined]  # ensured by is_applied_as_fieldop
```

Comment:
After #1994 the type-ignore might not be needed anymore. Please, merge back main and check.